Call volunteer logic

fifimova opened the pull request:
- TicketServiceImpl метод, который создает тикет  и присваивает его менее загруженному волонтеру
- UserRepo добавлен метод с запросом в бд, который ищет волонтеров с наименьшим числом тикетов
- UpdateHubServiceImpl добавлен метод, который обрабатывает коллбэк позвать волонтера и просить написать описание проблемы, метод добавлен в интерфес сервиса, чтобы можно было вызвать в любое меню
- исправила скрип таблицы тикета, там айдишник сам не инкрементился

1237633 committed:
Переместил клавы

fifimova committed:
- TicketServiceImpl метод, который создает тикет  и присваивает его менее загруженному волонтеру
- UserRepo добавлен метод с запросом в бд, который ищет волонтеров с наименьшим числом тикетов
- UpdateHubServiceImpl добавлен метод, который обрабатывает коллбэк позвать волонтера и просить написать описание проблемы, метод добавлен в интерфес сервиса, чтобы можно было вызвать в любое меню

1237633 commented on src/main/java/ru/kaInc/shelterbot/model/Ticket.java:
```diff
@@ -27,9 +27,13 @@ public class Ticket {
     @Column(name = "username")
     private String creatorsUsername;
 
+    @Column(name = "error_description")
```
предлагаю issue или problem description
Т.к. тикеты могут создаваться и системой

1237633 commented on src/main/java/ru/kaInc/shelterbot/service/implementation/UpdateHubServiceImpl.java:
```diff
@@ -119,9 +130,7 @@ public void processStart(Update update, List<Update> updates, TelegramBot telegr
-            SendMessage message = new SendMessage(update.message().chat().id(), "Айнц - цвай - драй - ничего не панимай");
-            telegramBot.execute(message);
-            keyboardBasic.processCommands(updates, telegramBot);
+            handleOtherCommands(update, telegramBot);
```
У нас же так и не происходит обработки колбека волонтера. Апдейт дальше передается в клавиатуры по очереди и все

1237633 commented on src/main/java/ru/kaInc/shelterbot/service/implementation/UserServiceImpl.java:
```diff
@@ -152,6 +153,26 @@ public User updateUser(User user) {
+    @Override
+    public User findAvailableVolunteer() {
+        List<User> volunteers = userRepo.findLeastBusyVolunteers(Role.VOLUNTEER, PageRequest.of(0, 1));
+        return volunteers.isEmpty() ? null : volunteers.get(0);
```
Может есть более подходящая структура данных? Например стек или очередь

fifimova replied:
внутри метода происходит логика по выбору волонтера у которого меньше всего тикетов, лист вполне врисовывается

1237633 commented on src/main/java/ru/kaInc/shelterbot/repo/UserRepo.java:
```diff
@@ -19,5 +22,13 @@ public interface UserRepo extends JpaRepository<User, Long> {
+            "GROUP BY u " +
+            "ORDER BY COUNT(t) ASC, u.id ASC")
+    List<User> findLeastBusyVolunteers(@Param("volunteerRole") Role volunteerRole, Pageable pageable);
+
```
Тут мы, насколько я виджу, пока не учитываем, что волонтер может быть не на работе, ему налетит 100500 тикетов и он вылетит в конец листа

fifimova replied:
просто волонтер у нас это и юзер, придётся ненужные поля городить
метод выбирает волонтера с наименьшим числом тикетов
в тикет можно добавить флаг решен/нерешен, и через апи волонтер будет его закрывать, соответсвенно меньше тикетов стало чем у других, допустим, и ему опять прилетит новый

fifimova committed:
- исправила нулпоинтер, который вылетал при создании тикета.
- тикет сохраняется в бд и присваивается волонтеру
- добавила данных в базу для проверки работоспособности, подвязала на волонтера свой акк

fifimova committed:
- api для получения всех тикетов, тикетов которые запросил пользователь, тикетов присвоенных волонтеру, для закрытия тикета

fifimova committed:
- небольшие правки для работы отправки оповещений волонтеру

fifimova committed:
- добавила логов
